tests/ops/op_math/test_evolve.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the ParametrizedEvolution class
+"""
+# pylint: disable=unused-argument, too-few-public-methods
+import pytest
+
+import pennylane as qml
+from pennylane.operation import AnyWires
+from pennylane.ops import Evolution, Evolve, QubitUnitary
+
+
+class MyOp(qml.RX):  # pylint: disable=too-few-public-methods
+    """Variant of qml.RX that claims to not have `adjoint` or a matrix defined."""
+
+    has_matrix = False
+    has_adjoint = False
+    has_decomposition = False
+    has_diagonalizing_gates = False
+
+
+def H(params, t, wires):
+    """Function that returns a time-dependent hamiltonian"""
+    ops = [
+        qml.PauliX(wires[0]),
+        qml.PauliZ(wires[1]),
+        qml.PauliY(wires[0]),
+        qml.PauliX(wires[1]),
+        qml.PauliZ(10),
+    ]
+    coeffs = [params[0], params[1], 4, 9, 10]
```

Comment:
Shouldn't we test the situation where some of the coefficients are callables that take `params, t`, rather than just an element from `params`? It seems like there is no time-dependence here.

Reply:
Added time-dependent tests!